include/xtensor/xhistogram.hpp

```diff
@@ -426,6 +442,60 @@ namespace xt
                         xt::ones<typename std::decay_t<E1>::value_type>(data.shape()),
                         minlength);
     }
+
+    /**
+     * Get the number of items in each bin, given the fraction of items per bin.
+     * The output is such that the total number of items of all bins is exactly "N".
+     *
+     * @param N the number of items to distribute
+     * @param weights fraction of items per bin: a 1D container whose size is the number of bins
+     *
+     * @return 1D container with the number of items per bin
+     */
+    template <class E>
+    inline xt::xtensor<size_t,1> bin_items(size_t N, E&& weights)
+    {
+        if (weights.size() <= 1) {
+            xt::xtensor<size_t,1> n = N * xt::ones<size_t>({1});
+            return n;
+        }
+
+        xt::xtensor<double,1> P = weights;
+
+        XTENSOR_ASSERT(xt::all(P >= 0.0));
+        XTENSOR_ASSERT(xt::sum(P)(0) > 0.0);
+
+        P /= xt::sum(P)(0);
```

Comment:
I think that a direct computation of P would be more efficient:
```cpp
xt::xtensor<double, 1> P = weights / xt::sum(P)(0);
```

The sassertions can be performed directly on `weights` (or on an adapter if `weights` can be a STL container).

Reply:
One has to be careful a bit to cast to `double` before dividing, otherwise the result is only zeros